packages/table/src/Row.tsx

```diff
@@ -0,0 +1,356 @@
+import React, { useState, useEffect } from 'react';
+import { Transition } from 'react-transition-group';
+import IconButton from '@leafygreen-ui/icon-button';
+import Icon from '@leafygreen-ui/icon';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import CheckboxCell from './CheckboxCell';
+import { useTableContext, Types, DataType } from './TableContext';
+
+const rowStyle = css`
+  border-top: 1px solid ${uiColors.gray.light2};
+  color: ${uiColors.gray.dark2};
+
+  & > td > div {
+    height: 40px;
+    overflow: hidden;
+    transition: all 150ms ease-in-out;
+  }
+`;
+
+const altColor = css`
+  &:nth-of-type(even) {
+    background-color: ${uiColors.gray.light3};
+  }
+`;
+
+const iconButtonMargin = css`
+  margin-right: 4px;
+`;
+
+const disabledStyle = css`
+  background-color: ${uiColors.gray.light2};
+  color: ${uiColors.gray.base};
+  cursor: not-allowed;
+  border-top: 1px solid ${uiColors.gray.light1};
+  border-bottom: 1px solid ${uiColors.gray.light1};
+`;
+
+const displayFlex = css`
+  display: flex;
+  align-items: center;
+`;
+
+const truncation = css`
+  max-width: 100px;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const transitionStyles = {
+  default: css`
+    transition: border 150ms ease-in-out;
+    border-top-color: transparent;
+
+    & > td {
+      padding-top: 0px;
+      padding-bottom: 0px;
+    }
+
+    & > td > div {
+      max-height: 0;
+    }
+  `,
+
+  entered: css`
+    border-top-color: ${uiColors.gray.light2};
+
+    & > td > div {
+      max-height: 40px;
+    }
+  `,
+};
+
+const styleMap = {
+  left: [DataType.String, DataType.Weight, DataType.ZipCode, DataType.Date],
+  right: [DataType.Number, DataType.Quantity],
+};
+
+function styleColumn(index: string, dataType: DataType) {
+  let justify = '';
+
+  if (styleMap.left.includes(dataType)) {
+    justify = 'flex-start';
+  } else if (styleMap.right.includes(dataType)) {
+    justify = 'flex-end';
+  }
+
+  return css`
+    & td:nth-child(${index}) > div {
+      justify-content: ${justify};
+    }
+  `;
+}
+
+function getIndentLevelStyle(indentLevel: number) {
+  return css`
+    & > td:nth-child(2) {
+      padding-left: ${8 + indentLevel * 16}px;
+    }
+  `;
+}
+
+function generateIndexRef() {
+  return Math.floor(Math.random() * 1000000 + 1);
```

Comment:
not that I'd ever hope for a table with tens of thousands of rows for the probability of collision to be feasible, but we could a larger set of possible doing something like `Math.random().toString(36).substring(2)`

Reply:
👍